### PR TITLE
Fix the '404 page not found' error when open a workspace

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Fix the '404 page not found' error when open a workspace

### What issues does this PR fix or reference?
fixes https://issues.redhat.com/browse/CRW-3864

### Is it tested? How?

1. Just create and open a new workspace:
Знімок екрана 2023-05-03 о 04 36 10
2. You will see an availability check before redirecting to the IDE.

### CODE REVIEW

File: packages/dashboard-frontend/src/services/helpers/ApiPing.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  }
+  let attempt = 0;
+  while (attempt < MAX_ATTEMPT_QUANTITY) {
+    try {
+      await axios.get(url);
+      return true;
+    } catch (error) {
+      if (
+        common.helpers.errors.includesAxiosResponse(error) &&
+        (error as AxiosError).response?.status === 404
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/services/helpers/ApiPing.ts:31]:
```suggestion
        common.helpers.errors.includesAxiosResponse(error) && error.response?.status === 404
```

Comment (olexii4) [reply packages/dashboard-frontend/src/services/helpers/ApiPing.ts]: I have fixed it.

File: packages/dashboard-frontend/src/services/helpers/__tests__/api-ping.spec.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    });
+
+    const isAvailable = await isAvailableEndpoint(endpoint);
+
+    expect(delayMock).not.toBeCalled();
+    expect(console.error).not.toBeCalled();
+    expect(isAvailable).toBeTruthy();
+  });
+
+  it('should return "true" if response status 503', async () => {
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/services/helpers/__tests__/api-ping.spec.ts:96]: Something wrong here: either the test title or the expectation.

File: packages/dashboard-frontend/src/services/helpers/__tests__/api-ping.spec.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+    expect(delayMock).toBeCalledWith(2500);
+    expect(delayMock).toHaveBeenCalledTimes(11);
+    expect(console.error).toBeCalledWith(
+      `Endpoint 'https://example.com/developer-image/3100/' is not available. Error: 503 Service Unavailable.`,
+    );
+    expect(isAvailable).toBeFalsy();
+  });
+
+  it('should return "true" if response status 404', async () => {
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/services/helpers/__tests__/api-ping.spec.ts:117]: The same here.

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**